LuckyDragon/app/src/main/java/com/example/luckydragon/Activities/ViewEventActivity.java

```diff
@@ -191,17 +194,46 @@ public void loadChildFragment() {
     public void sampleAttendeesIfNeccessary() {
         if(!event.isLoaded()) return;
         if(event == null || event.getCreatedTimeMillis() == null) return;
+
+        GlobalApp globalApp = (GlobalApp) getApplication();
         if(!event.haveInviteesBeenSelected()) {
             LocalDateTime currentDateTime = LocalDateTime.now();
             LocalDate lotteryDate = LocalDate.parse(event.getLotteryDate());
             LocalTime lotteryTime = LocalTime.of(event.getLotteryHours(), event.getLotteryMinutes());
             LocalDateTime lotteryDateTime = LocalDateTime.of(lotteryDate, lotteryTime);
             if(currentDateTime.isAfter(lotteryDateTime)) {
                 event.selectInviteesFirstTime();
+
+                String notSelectedTitle = "Update re - " + event.getName();
+                String notSelectedBody = "You have unfortunately not been selected to attend " +
+                        "the '" + event.getName() + "' event. You will remain on the waitlist " +
+                        "and may possibly be selected if another entrant declines to attend.";
+                for (String deviceId : event.getWaitList()) {
+                    User notSelectedUser = globalApp.getUserById(deviceId);
+                    notSelectedUser.addToNotificationList(notSelectedTitle, notSelectedBody);
+                }
+
+                String selectedTitle = "Congratulations!";
+                String selectedBody = "You have been selected to attend the '" + event.getName() +
+                        "' event! Please open the event in the app and accept the invitation " +
+                        "for the event if you would like to attend.";
+                for (String deviceId : event.getInviteeList()) {
+                    User selectedUser = globalApp.getUserById(deviceId);
+                    selectedUser.addToNotificationList(selectedTitle, selectedBody);
+                }
```

Comment:
Put this in a function please. Make it part of something like Event or NotificationList or NotificationService

Reply:
The only thing gained by putting this in a function is removing the loop, which likely won’t save any lines of code at all since we are saving literally 1 line.

Reply:
You should put it in the selectInvitees function because the code here always should occur when you call selectInvitees regardless of what activity you do it in.

Reply:
You should put it in the selectInvitees function because the code here always should occur when you call selectInvitees regardless of what activity you do it in. See the comment below for implementation details

Reply:
But there would still be 3 different cases of sending notifs:
1. User wins lottery
2. User loses lottery
3. User is selected as replacement

In selectInvitees, you have no knowledge of which of these cases it is (and can’t even do Case 2). So you’d still have to identify separate areas of where to send these 3 cases, so the most readable solution is having it in the activity.

Reply:
1. Have utility functions for sending notifications to the entire waitlist, invitee list etc. that take in as param, a title and body.
2. Use these functions for here and in organizerNotificationDialog.
3. Call the utilty function for wiatlist and invitee list after the current code in selectInvitees
4. Now if in the future we need to selectinvitees not on EventActivity startup, we can easily move it.